sources/TileDB.CSharp/FragmentInfo.cs

```diff
@@ -0,0 +1,660 @@
+        public bool IsSparse(uint fragmentIndex)
+        {
+            using var ctxHandle = _ctx.Handle.Acquire();
+            using var handle = _handle.Acquire();
+            int result;
+            _ctx.handle_error(Methods.tiledb_fragment_info_get_sparse(ctxHandle, handle, fragmentIndex, &result));
+            return result == 1;
+        }
+
+        /// <summary>
+        /// Gets the timestamp range of a fragment.
+        /// </summary>
+        /// <param name="fragmentIndex">The index of the fragment of interest.</param>
+        /// <returns>The start and end timestamps of the fragment.</returns>
+        /// <remarks>
+        /// The maximum value <paramref name="fragmentIndex"/> can take
+        /// is determined by the <see cref="FragmentCount"/> property.
+        /// </remarks>
+        public (ulong Start, ulong End) GetTimestampRange(uint fragmentIndex)
+        {
+            using var ctxHandle = _ctx.Handle.Acquire();
+            using var handle = _handle.Acquire();
+            ulong start, end;
+            _ctx.handle_error(Methods.tiledb_fragment_info_get_timestamp_range(ctxHandle, handle, fragmentIndex, &start, &end));
+            return (start, end);
+        }
+
+        /// <summary>
+        /// Gets the number of Minimum Bounded Rectangles (MBRs) of a fragment.
+        /// </summary>
+        /// <param name="fragmentIndex">The index of the fragment of interest.</param>
+        /// <remarks>
+        /// The maximum value <paramref name="fragmentIndex"/> can take
+        /// is determined by the <see cref="FragmentCount"/> property.
+        /// </remarks>
+        public ulong GetMinimumBoundedRectangleCount(uint fragmentIndex)
+        {
+            using var ctxHandle = _ctx.Handle.Acquire();
+            using var handle = _handle.Acquire();
+            ulong result;
+            _ctx.handle_error(Methods.tiledb_fragment_info_get_mbr_num(ctxHandle, handle, fragmentIndex, &result));
+            return result;
+        }
+
+        /// <summary>
+        /// Gets the Minimum Bounded Rectangle (MBR) from one of a fragment's dimensions, identified by its index.
+        /// </summary>
+        /// <typeparam name="T">The dimension's data type.</typeparam>
+        /// <param name="fragmentIndex">The index of the fragment of interest.</param>
+        /// <param name="minimumBoundedRectangleIndex">The index of the MBR of interest.</param>
+        /// <param name="dimensionIndex">The index of the dimension of interest, following
+        /// the order as it was defined in the domain of the array schema.</param>
+        /// <returns>The start and end values of the MBR, inclusive.</returns>
+        /// <exception cref="NotSupportedException"><typeparamref name="T"/> is a managed type
+        /// other than <see cref="string"/>.</exception>
+        /// <remarks>
+        /// In case of a type mismatch this function will not fail but return invalid data.
+        /// </remarks>
```

Comment:
Thoughts about this? To check the generic type we would have to allocate an `ArraySchema`; the function currently does not allocate. Is it OK to return garbage if the user specifies the wrong type? This is an advanced API either way.

Reply:
Done.